File: database-driver-api/src/main/java/de/lino/database/database/exception/NoSuchDataFound.java
package de.lino.database.database.exception;

import org.jetbrains.annotations.NotNull;

/**
 * Thrown when a stored record is found without the expected {@code "data"} payload while a
 * {@link de.lino.database.database.DatabaseSection} is loading or reading its entries, indicating
 * corrupted or unexpectedly shaped persisted data.
 */
public class NoSuchDataFound extends RuntimeException {

    /**
     * Creates a new exception for the record with the given id that is missing its data payload.
     *
     * @param id the primary key of the affected record
     */
    public NoSuchDataFound(@NotNull String id) {
        super("No such data found in document");
    }

}

File: python/database-driver-api/src/database_driver/api/database/exception/no_such_data_found.py
"""Mirror of ``de.lino.database.database.exception.NoSuchDataFound``."""

from __future__ import annotations


class NoSuchDataFound(RuntimeError):
    """Raised when a stored record is found without the expected ``"data"`` payload while
    a ``DatabaseSection`` is loading or reading its entries, indicating corrupted or
    unexpectedly shaped persisted data."""

    def __init__(self, id: str) -> None:
        # The Java original deliberately omits the id from the message; kept identical so
        # log output stays comparable between the two editions.
        super().__init__("No such data found in document")
